sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -2349,6 +2351,46 @@ def __GetBodiesFromQueryResult(result):
         # Query operations will use ReadEndpoint even though it uses POST(for regular query operations)
         request_params = _request_object.RequestObject(typ, documents._OperationType.SqlQuery)
         req_headers = base.GetHeaders(self, initial_headers, "post", path, id_, typ, options, partition_key_range_id)
+
+        # check if query has prefix partition key
+        cont_prop = kwargs.pop("containerProperties", None)
+        partition_key = options.get("partitionKey", None)
+        isPrefixPartitionQuery = False
+        partition_key_definition = None
+        if cont_prop:
+            cont_prop = await cont_prop()
+            pk_properties = cont_prop["partitionKey"]
+            partition_key_definition = PartitionKey(path=pk_properties["paths"], kind=pk_properties["kind"])
+            if partition_key_definition.kind != "MultiHash":
+                isPrefixPartitionQuery = False
+            elif type(partition_key) == list and len(partition_key_definition['paths']) == len(partition_key):
+                isPrefixPartitionQuery = False
+            else:
+                isPrefixPartitionQuery = True
+
+        if isPrefixPartitionQuery:
+            # here get the overlap, then do one of two scenarios
+            req_headers.pop(http_constants.HttpHeaders.PartitionKey, None)
+            feedrangeEPK = partition_key_definition._get_epk_range_for_prefix_partition_key(partition_key)  # cspell:disable-line # pylint: disable=line-too-long
+            over_lapping_ranges = await self._routing_map_provider.get_overlapping_ranges(id_, [feedrangeEPK])
+
+            if over_lapping_ranges:
+                # For epk range filtering we can end up in one of 2 cases:
```

Comment:
What happens if the EPK Range spans multiple physical partitions? In the case of a Partial PK, they can span more than 1 partition

Reply:
Did we test this with a container where this happens? (I think I gave you the pointers on how to test this, if not, please ping me)

Reply:
Yes We did this test this. In the python sdk routing was already being handled, in this case the solution was implemented before we needed it thankfully. self._routing_map_provider.get_overlapping_ranges(id_, [feedrangeEPK]) this specific line goes into the python sdk routing map implementation and in there it will handle the case of their being multiple physical partitions before we get to the next part.

Reply:
Did we specifically test the scenario where the partial PK returned 2 partitions?
How is the handling exactly? When I look at https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py#L113 doesn't that return an array of 2 items?

Reply:
> Did we specifically test the scenario where the partial PK returned 2 partitions? How is the handling exactly? When I look at
> 
> https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/azure/cosmos/_routing/collection_routing_map.py#L113
> 
> doesn't that return an array of 2 items?

When the overlap includes more than partition that means there was a partition split, that scenario is tested for here:
[https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/test/routing/test_collection_routing_map.py#L44](https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/test/routing/test_collection_routing_map.py#L44)

that method will only return a list with a single element.

Reply:
In the case that there are two overlapping partitions it means theres a partition split to account for. That is tested for in the routing tests we have:
[routing test](https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/test/routing/test_collection_routing_map.py#L44)

Also the get overlapping ranges that is used is actually this one:
[smart routing](https://github.com/Azure/azure-sdk-for-python/blob/d3ee19bc26e579e1b91fa6a651a4ce8de158edaf/sdk/cosmos/azure-cosmos/azure/cosmos/_routing/routing_map_provider.py#L135)

Reply:
> When the overlap includes more than partition that means there was a partition split, that scenario is tested for here:

This is incorrect. A Partial Partition Key can span many physical partitions and there are no splits involved.

Reply:
GetOverlappingRanges should return 2+ items and whoever is executing the operation should handle it.

There are operations for which this is unsupported, for example, CRUD of items, or TransactionalBatch, but others where it makes sense, like Query.

Reply:
I adjusted the prefix query code to handle multiple over lapping ranges. It will now query items from all the partitions that the partial partition key spans.